tests/server/cluster/cluster_test.go

```diff
@@ -0,0 +1,750 @@
+	// When store is up:
+	{
+		// Case 1: RemoveStore should be OK;
+		resetStoreState(c, rc, store.GetId(), metapb.StoreState_Up)
+		err := rc.RemoveStore(store.GetId())
+		c.Assert(err, IsNil)
+		removedStore := getStore(c, clusterID, grpcPDClient, store.GetId())
+		c.Assert(removedStore.GetState(), Equals, metapb.StoreState_Offline)
+		// Case 2: BuryStore w/ force should be OK;
+		resetStoreState(c, rc, store.GetId(), metapb.StoreState_Up)
+		err = rc.BuryStore(store.GetId(), true)
+		c.Assert(err, IsNil)
+		buriedStore := getStore(c, clusterID, grpcPDClient, store.GetId())
+		c.Assert(buriedStore.GetState(), Equals, metapb.StoreState_Tombstone)
+		// Case 3: BuryStore w/o force should fail.
+		resetStoreState(c, rc, store.GetId(), metapb.StoreState_Up)
+		err = rc.BuryStore(store.GetId(), false)
+		c.Assert(err, NotNil)
+	}
+
+	// When store is offline:
+	{
+		// Case 1: RemoveStore should be OK;
+		resetStoreState(c, rc, store.GetId(), metapb.StoreState_Offline)
+		err := rc.RemoveStore(store.GetId())
+		c.Assert(err, IsNil)
+		removedStore := getStore(c, clusterID, grpcPDClient, store.GetId())
+		c.Assert(removedStore.GetState(), Equals, metapb.StoreState_Offline)
+		// Case 2: BuryStore w/ or w/o force should be OK.
+		resetStoreState(c, rc, store.GetId(), metapb.StoreState_Offline)
+		err = rc.BuryStore(store.GetId(), false)
+		c.Assert(err, IsNil)
+		buriedStore := getStore(c, clusterID, grpcPDClient, store.GetId())
+		c.Assert(buriedStore.GetState(), Equals, metapb.StoreState_Tombstone)
+	}
+
+	// When store is tombstone:
+	{
+		// Case 1: RemoveStore should should fail;
+		resetStoreState(c, rc, store.GetId(), metapb.StoreState_Tombstone)
+		err := rc.RemoveStore(store.GetId())
+		c.Assert(err, NotNil)
+		// Case 2: BuryStore w/ or w/o force should be OK.
+		resetStoreState(c, rc, store.GetId(), metapb.StoreState_Tombstone)
+		err = rc.BuryStore(store.GetId(), false)
+		c.Assert(err, IsNil)
+		buriedStore := getStore(c, clusterID, grpcPDClient, store.GetId())
+		c.Assert(buriedStore.GetState(), Equals, metapb.StoreState_Tombstone)
+	}
+
+	{
+		// Put after removed should return tombstone error.
+		resp, err := putStore(c, grpcPDClient, clusterID, store)
+		c.Assert(err, IsNil)
+		c.Assert(resp.GetHeader().GetError().GetType(), Equals, pdpb.ErrorType_STORE_TOMBSTONE)
+	}
+	{
+		// Update after removed should return tombstone error.
+		req := &pdpb.StoreHeartbeatRequest{
+			Header: testutil.NewRequestHeader(clusterID),
+			Stats:  &pdpb.StoreStats{StoreId: store.GetId()},
+		}
+		resp, err := grpcPDClient.StoreHeartbeat(context.Background(), req)
+		c.Assert(err, IsNil)
+		c.Assert(resp.GetHeader().GetError().GetType(), Equals, pdpb.ErrorType_STORE_TOMBSTONE)
+	}
+}
+
+// Make sure PD will not panic if it start and stop again and again.
+func (s *clusterTestSuite) TestRaftClusterRestart(c *C) {
+	tc, err := tests.NewTestCluster(1)
```

Comment:
Does it no longer need the previous `cleanup`?

Reply:
I think we can use `Destroy` to clean up.